src/renderer/src/pages/settings/settings-general.tsx

```diff
@@ -15,34 +18,41 @@ export function SettingsGeneral({
   userPreferences,
   updateUserPreferences,
 }: SettingsGeneralProps) {
+  const { t } = useTranslation("settings");
+
   const [form, setForm] = useState({
     downloadsPath: "",
     downloadNotificationsEnabled: false,
     repackUpdatesNotificationsEnabled: false,
+    language: "",
   });
 
+  const [defaultDownloadsPath, setDefaultDownloadsPath] = useState("");
+
   useEffect(() => {
-    if (userPreferences) {
-      const {
-        downloadsPath,
-        downloadNotificationsEnabled,
-        repackUpdatesNotificationsEnabled,
-      } = userPreferences;
-
-      window.electron.getDefaultDownloadsPath().then((defaultDownloadsPath) => {
-        setForm((prev) => ({
-          ...prev,
-          downloadsPath: downloadsPath ?? defaultDownloadsPath,
-          downloadNotificationsEnabled,
-          repackUpdatesNotificationsEnabled,
-        }));
-      });
+    async function fetchdefaultDownloadsPath() {
+      setDefaultDownloadsPath(await window.electron.getDefaultDownloadsPath());
     }
-  }, [userPreferences]);
 
-  const { t } = useTranslation("settings");
+    fetchdefaultDownloadsPath();
+  }, []);
+
+  useEffect(updateFormWithUserPreferences, [
+    userPreferences,
+    defaultDownloadsPath,
+  ]);
+
+  const handleLanguageChange = (event) => {
+    const value = event.target.value;
+
+    handleChange({ language: value });
+    changeLanguage(value);
+  };
 
   const handleChange = (values: Partial<typeof form>) => {
+    // TODO: why is the setForm needed if updateUserPreferences already changes the
```

Comment:
Could you just remove these comments?

Reply:
is there already an issue about this? if not, i can create one and remove the comment

Reply:
There's not, you may create one